### PR TITLE
CS-868 Add Luhn Algorithm

### PR DESCRIPTION
**Pull request by ShaPhi7**

### CODE REVIEW

File: algorithms-modules/pom.xml [hunk 1]
```diff
@@ -21,6 +21,7 @@
         <module>algorithms-miscellaneous-4</module>
         <module>algorithms-miscellaneous-5</module>
         <module>algorithms-miscellaneous-6</module>
+		<module>algorithms-miscellaneous-7</module>
```
Comment (Bwvolleyball) [inline algorithms-modules/pom.xml:24]: nit, the spacing of this module entry looks a bit off

File: algorithms-modules/algorithms-miscellaneous-7/src/main/java/com/baeldung/algorithms/luhn/LuhnChecker.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
+package com.baeldung.algorithms.luhn;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LuhnChecker {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(LuhnChecker.class);
```
Comment (KevinGilmore) [inline algorithms-modules/algorithms-miscellaneous-7/src/main/java/com/baeldung/algorithms/luhn/LuhnChecker.java:8]: Use spaces to indent

File: algorithms-modules/algorithms-miscellaneous-7/src/test/java/com/baeldung/algorithms/luhn/LuhnCheckerUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+public class LuhnCheckerUnitTest {
+
+    @Test
+    public void whenCardNumberDoesMeetLuhnCriteria_thenCheckLuhnReturnsTrue() {
+
+    	String cardNumber = "8649";
+		boolean result = LuhnChecker.checkLuhn(cardNumber);
```
Comment (KevinGilmore) [inline algorithms-modules/algorithms-miscellaneous-7/src/test/java/com/baeldung/algorithms/luhn/LuhnCheckerUnitTest.java:12]: Indent with spaces. (This occurs in the other methods below, too)

File: algorithms-modules/algorithms-miscellaneous-7/src/main/java/com/baeldung/algorithms/luhn/LuhnChecker.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+     * This is how the final digit of a credit card is calculated.
+     */
+    public static boolean checkLuhn(String cardNumber) {
+        int sum = 0;
+
+        try {
+
+            for (int i = cardNumber.length() - 1; i >= 0; i--) {
+
+                int digit = Integer.parseInt(cardNumber.substring(i, i + 1));
```
Comment (Bwvolleyball) [inline algorithms-modules/algorithms-miscellaneous-7/src/main/java/com/baeldung/algorithms/luhn/LuhnChecker.java:28]: I think we need to tighten up this spacing as well

```suggestion
        try {
            for (int i = cardNumber.length() - 1; i >= 0; i--) {
                int digit = Integer.parseInt(cardNumber.substring(i, i + 1));
```

**Pull request merged**